packages/relay/src/lib/eth.ts

```diff
@@ -764,9 +764,21 @@ export class EthImpl implements Eth {
   async getBlockByNumber(blockNumOrTag: string, showDetails: boolean, requestId?: string): Promise<Block | null> {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     this.logger.trace(`${requestIdPrefix} getBlockByNumber(blockNum=${blockNumOrTag}, showDetails=${showDetails})`);
-    return this.getBlock(blockNumOrTag, showDetails, requestId).catch((e: any) => {
-      throw this.genericErrorHandler(e, `${requestIdPrefix} Failed to retrieve block for blockNum ${blockNumOrTag}`);
-    });
+
+    const cacheKey = `eth_getBlockByNumber_${blockNumOrTag}_${showDetails}`;
+    let block = this.cache.get(cacheKey);
+    if (!block) {
+      block = this.getBlock(blockNumOrTag, showDetails, requestId).catch((e: any) => {
+        throw this.genericErrorHandler(e, `${requestIdPrefix} Failed to retrieve block for blockNum ${blockNumOrTag}`);
+      });
+
+      if (blockNumOrTag != EthImpl.blockLatest && blockNumOrTag != EthImpl.blockPending) {
```

Comment:
Is the case when blockNumOrTag === null handled?

Reply:
It's always a string (earliest/pending/latest) or hex (function param `blockNumOrTag: string`), and it can not be `null`. The validation will throw an exception if anyone passes `null`.

Reply:
Ah I see.  I guess we can get rid of the checks for null then in `getHistoricalBlockResponse` and `EthImpl.blockTagIsLatestOrPending`?

Reply:
Please add this as a comment in the code so it's clear

Reply:
@lukelee-sl There are paths where we need these checks, and can't completely rid of the null validation.
@Nana-EC Added + some tests.